File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:

        def helper(s,left, right):
            count = 0
            while left >= 0 and right < len(s) and s[left] == s[right]:
                count += 1
                left -= 1
                right += 1
            
            return count 

        result = 0

        for index in range(len(s)):
            result += helper(s, index, index)
            result += helper(s, index, index + 1)

        return result

  

File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
#include <vector>
#include <algorithm> // For std::max

using namespace std;

class Solution {
public:
    int trap(vector<int>& height) {
        // Handle the edge case where the map is empty
        if (height.empty()) {
            return 0;
        }

        int result = 0;
        
        // Initialize our two pointers
        int left = 0;
        int right = height.size() - 1;

        // Initialize the max heights seen from both sides
        int lmax = height[left];
        int rmax = height[right];

        // Process the array until the two pointers meet
        while (left < right) {
            // The smaller of the two maxes bottlenecks the water
            if (lmax < rmax) {
                left++; // Move the left pointer inward
                lmax = max(lmax, height[left]); // Update the left max
                result += lmax - height[left];  // Add trapped water
            } else {
                right--; // Move the right pointer inward
                rmax = max(rmax, height[right]); // Update the right max
                result += rmax - height[right];  // Add trapped water
            }
        }

        return result;
    }
};